src/log/zipkin.cpp

```diff
@@ -155,13 +168,36 @@ fc::variant create_zipkin_variant( zipkin_span::span_data&& span, const std::str
    return result;
 }
 
+void zipkin::post_request(zipkin_span::span_data&& span) {
+   boost::asio::post(my->work_strand, [this, span{std::move(span)}]() mutable {
+         my->log( std::move( span ) );
+   });
+}
+
 void zipkin::log( zipkin_span::span_data&& span ) {
-   if( my->consecutive_errors > my->max_consecutive_errors || my->stopped )
+   if( my->stopped ) {
       return;
+   }else if( fc::do_sighup && fc::sighup_requested.load()) {
+      fc::do_sighup = 0;
+      fc::sighup_requested = false;
+      my->consecutive_errors = 0;
+   }else if( my->consecutive_errors > my->max_consecutive_errors ) {
+      return;
+   }
 
-   boost::asio::post(my->work_strand, [this, span{std::move(span)}]() mutable {
-      my->log( std::move( span ) );
-   });
+   if( my->consecutive_errors > 0 ) {
+      if( my->timer_expired ) {
+         my->timer_expired = false;
```

Comment:
`timer_expired` is not thread safe.

Reply:
Changed the definition of `timer_expired` to make it thread safe.